src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1636,7 +1636,7 @@ private CodeClass AddModelClass(OpenApiUrlTreeNode currentNode, OpenApiSchema sc
             {
                 DocumentationLabel = schema.ExternalDocs?.Description ?? string.Empty,
                 DocumentationLink = schema.ExternalDocs?.Url,
-                Description = schema.Description.CleanupDescription(),
+                Description = (string.IsNullOrEmpty(schema.Description) ? schema.AllOf.LastOrDefault()?.Description : schema.Description).CleanupDescription(),
```

Comment:
@baywet Are you sure this change is correct? This seems to result in incorrect descriptions for some types in the Graph SDK. For example, `microsoft.graph.omaSettingBase64` gets `OMA Settings definition.`, which is from `microsoft.graph.omaSetting`. With the old code, it was `OMA Settings Base64 definition.`, which is correct.

Reply:
Thank you for bringing this up. Yes after giving it a second thought I guess it should have the filter clause in the last or default. Sorry about that. Would you mind fixing this and adding a unit test to avoid regressions?